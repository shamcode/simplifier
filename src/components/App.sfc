<template>
    {% import Header from './Header.sht' %}
    {% import Footer from './Footer.sht' %}
    {% import Form from './Form.sfc' %}
    {% import DisplayResult from './DisplayResult.sht' %}

    <Header/>
    <main>
        <Form
            onSubmit={{onSubmit}}
        />
        <DisplayResult
            result={{result}}
        />
    </main>
    <Footer/>
</template>

<script>
    import simplify from 'logic-expression-simplify';

    function App( options, update ) {
        const result = ref();

        options( {
            [ result ]: '',
            [ ref( 'onSubmit' ) ]( evt, { expression } ) {
                evt.preventDefault();

                let simplifiedExpression;
                try {
                    simplifiedExpression = simplify( expression );
                } catch ( e ) {
                    simplifiedExpression = '';
                }
                update( {
                    [ result ]: simplifiedExpression
                } );

                return false;
            }
        } );
    }

    export default Component( Template, App );
</script>
