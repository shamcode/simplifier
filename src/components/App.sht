{% import Header from './Header.sht' %}
{% import Footer from './Footer.sht' %}
{% import Form from './Form.sfc' %}
{% import DisplayResult from './DisplayResult.sht' %}

<Header/>
<main>
    <Form
        onSubmit={{this.options.onSubmit.bind( this )}}
    />
    <DisplayResult result={{result}}/>
</main>
<Footer/>
