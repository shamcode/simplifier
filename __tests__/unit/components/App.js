import App from '../../../src/components/App.sht';
import { ref, onsubmit } from 'sham-ui-directives';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    expect.assertions( 3 );
    const fn = jest.fn();

    const meta = renderer( App, {
        directives: {
            ref,
            onsubmit
        },
        result: 'a || b',
        onSubmit: fn
    } );

    meta.component.container.querySelector( 'input' ).value = 'a && b';
    meta.component.container.querySelector( 'form' ).dispatchEvent( new Event( 'submit' ) );

    const json = meta.toJSON();
    delete json.Options.onSubmit;
    expect( json ).toMatchSnapshot();
    expect( fn.mock.calls ).toHaveLength( 1 );
    expect( fn.mock.calls[ 0 ][ 1 ] ).toEqual( { expression: 'a && b' } );
} );
