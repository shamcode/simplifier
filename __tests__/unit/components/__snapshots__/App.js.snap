// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Constructor": "App",
  "Options": Object {
    "ID": "component",
    "containerSelector": "body",
    "directives": Object {
      "onsubmit": [Function],
      "ref": [Function],
    },
    "result": "a || b",
    "types": Array [],
  },
  "html": "
<header>
  <h1>Logic expressions simplifier</h1>
</header>
<!--Header-->
<main>
  <form>
    <input id=\\"input-text\\" placeholder=\\"(a &amp;&amp; b) || (!a &amp;&amp; !b)\\" tabindex=\\"1\\" name=\\"expression\\">
    <!--Input-->
    <button type=\\"submit\\" tabindex=\\"2\\">Simplify!</button>
  </form>
  <!--Form-->
  <div id=\\"result\\">a || b</div>
  <!--DisplayResult-->
</main>
<footer></footer>
<!--Footer-->
",
}
`;
